lib/auth/backends/ChainBackend.js

```diff
@@ -0,0 +1,187 @@
+            scopeDate,
+            options,
+            done
+        ), callback);
+    }
+
+    static _mergeObjects(objectResponses) {
+        return objectResponses.reduce(
+            (retObj, resObj) => Object.assign(retObj, resObj.message.body),
+            {});
+    }
+
+    getCanonicalIds(emailAddresses, options, callback) {
+        this._forEachClient(
+            (client, done) => client.getCanonicalIds(emailAddresses, options, done),
+            (err, res) => {
+                if (err) {
+                    return callback(err);
+                }
+                // TODO: atm naive merge, better handling of conflicting email results
+                return callback(null, {
+                    message: {
+                        body: ChainBackend._mergeObjects(res),
+                    },
+                });
+            });
+    }
+
+    getEmailAddresses(canonicalIDs, options, callback) {
+        this._forEachClient(
+            (client, done) => client.getEmailAddresses(canonicalIDs, options, done),
+            (err, res) => {
+                if (err) {
+                    return callback(err);
+                }
+                return callback(null, {
+                    message: {
+                        body: ChainBackend._mergeObjects(res),
+                    },
+                });
+            });
+    }
+
+    /*
+     * merge policy responses into a single message
+     */
+    static _mergePolicies(policyResponses) {
+        const policyMap = {};
+
+        policyResponses.forEach(resp => {
+            if (!resp.message || !Array.isArray(resp.message.body)) {
+                return;
+            }
+
+            resp.message.body.forEach(policy => {
+                const arn = policy.arn || '';
+                if (!policyMap[arn]) {
+                    policyMap[arn] = policy.isAllowed;
+                }
+                // else is duplicate policy
+            });
+        });
+
+        const policyList = Object.keys(policyMap).map(arn => {
+            if (arn === '') {
+                return { isAllowed: policyMap[arn] };
+            }
+            return { isAllowed: policyMap[arn], arn };
+        });
+
+        return policyList;
+    }
+
+    /*
+        response format:
+            {  message: {
+                body: [{}],
+                code: number,
+                message: string,
+            } }
+     */
+    checkPolicies(requestContextParams, userArn, options, callback) {
+        this._forEachClient((client, done) => client.checkPolicies(
+            requestContextParams,
+            userArn,
+            options,
+            done
+        ), (err, res) => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, {
+                message: {
+                    body: ChainBackend._mergePolicies(res),
+                },
+            });
+        });
+    }
+
+    healthcheck(reqUid, callback) {
```

Comment:
@naren-scality @RachedBenMustapha updated the healthcheck to test all clients and not exit on early error.